hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryBasicTest.java

```diff
@@ -952,6 +954,35 @@ public void testQueryPortableObjectWithIndexAndAlwaysCacheValues() {
         testQueryUsingPortableObject(config, name);
     }
 
+    @Test
+    public void testIndexesOnKeyAttributes() {
+        String name = randomMapName();
+        Config config = smallInstanceConfig();
+        config.getMapConfig(name)
+                .addMapIndexConfig(new MapIndexConfig("__key.a", false))
+                .addMapIndexConfig(new MapIndexConfig("__key#b", true));
+
+        HazelcastInstance instance = createHazelcastInstance(config);
+        IMap<CompositeKey, Integer> map = instance.getMap(name);
+        map.addIndex("__key.c", true);
+        map.addIndex("__key#d", false);
+
+        for (int i = 0; i < 10; i++) {
+            map.put(new CompositeKey(i), i);
+        }
+
+        map.values(Predicates.equal("__key.a", 5));
+        map.values(Predicates.equal("__key#b", 5));
+        map.values(Predicates.equal("__key#c", 5));
+        map.values(Predicates.equal("__key.d", 5));
+
+        Map<String, LocalIndexStats> stats = map.getLocalMapStats().getIndexStats();
+        assertEquals(1, stats.get("__key.a").getQueryCount());
```

Comment:
I would also check that before queries the `stats` values are zero.

Reply:
done